### ISSUE TITLE
Can't create a displaced array

### ISSUE THREAD

**Issue by stassats** [drmeister/clasp#127]
Starting Boehm cclasp 0.3-test-9-2-g31e57fd ... loading image... it takes a few seconds
Top level.
> (make-array 2 :displaced-to #(1 2 3))

Condition of type: SIMPLE-TYPE-ERROR
In lisp_errorUnexpectedType src/core/foundation.cc line 158
COMMON-LISP:TYPE-ERROR :initializers (:DATUM COMMON-LISP:NIL :EXPECTED-TYPE COMMON-LISP:INTEGER  )

**Comment by Shinmera** [comment]
Confirmed on b05884f
```
> (make-array 2 :displaced-to #(1 2 3))

Condition of type: SIMPLE-TYPE-ERROR
In lisp_errorUnexpectedType src/core/foundation.cc line 161
COMMON-LISP:TYPE-ERROR :initializers (:DATUM COMMON-LISP:NIL :EXPECTED-TYPE COMMON-LISP:INTEGER  )
```

**Comment by Shinmera** [comment]
While it seems fixed on 482d6ca5bd8ad30ff8c9b6a835989f3aec395e1a
```
> (make-array 2 :displaced-to #(1 2 3))

#<array 0x7f453d120199>
```
it's still broken in other regards, most obviously the printing issue also shown in #125, but then there's also this:
```
> (elt * 0)

Condition of type: SIMPLE-TYPE-ERROR
In cl_elt ../../src/core/sequence.cc line 73
COMMON-LISP:TYPE-ERROR :initializers (:DATUM #1A(1 2 ) :EXPECTED-TYPE COMMON-LISP:SEQUENCE  )


Available restarts:
(use :r1 to invoke restart 1)

1. (RESTART-TOPLEVEL) Go back to Top-Level REPL.

Broken at frame[115] TOP-LEVEL.
 File: "/opt/clasp/build/clasp/build/clasp/Contents/Resources/lisp/kernel/lsp/top.lsp" (Position #21032)
```

**Comment by Shinmera** [comment]
Still around in all their buggy glory on 62a597141aa08d54ae16277cd83c191ad447c3b6

**Comment by Shinmera** [comment]
Additional issues:
```
(length (make-array 2 :displaced-to #(a b c)))
(typep (make-array 2 :displaced-to #(a b c)) 'sequence)
```

**Comment by Shinmera** [comment]
Fixed on 31e785daa74e72b7c9048a3a32e4cf03bfd97688
```
> (elt (make-array 2 :displaced-to #(1 2 3)) 0)

1
> (aref (make-array 2 :displaced-to #(1 2 3)) 2)

Condition of type: SIMPLE-PROGRAM-ERROR
Index too large 2 must be less than 2

Available restarts:
(use :r1 to invoke restart 1)

1. (RESTART-TOPLEVEL) Go back to Top-Level REPL.

Broken at frame[116] TOP-LEVEL.
 File: #P"/media/autobuild/linut/clasp-testing/.autobuild/31e785daa74e72b7c9048a3a32e4cf03bfd97688/build/clasp/Contents/Resources/lisp/kernel/lsp/top.lsp" (Position #21029)
>> :r1

> (length (make-array 2 :displaced-to #(1 2 3)))

2
> (typep (make-array 2 :displaced-to #(1 2 3)) 'sequence)

T
```
